drs_cli/client.py

```diff
@@ -1,229 +1,424 @@
 """Class implementing DRS client."""
 
+import json
 import logging
+import re
 import requests
-from typing import (Dict, Optional, Union)
-
-from drs_cli.utils import (
-    check_drs_uri_regex,
-    check_http_uri_regex,
-    check_https_uri_regex,
-    check_drs_id_regex,
-    get_drs_uri_host_and_id,
-    get_http_uri_host,
-    get_https_uri_host
-)
+import socket
+import sys
+from typing import (Dict, Optional, Tuple, Union)
+from urllib.parse import quote
+import urllib3
+
+import pydantic
+
 from drs_cli.models import (AccessURL, DrsObject, Error, PostDrsObject)
-from drs_cli.errors import DRS_URI_FORMAT_ERROR
+from drs_cli.errors import (
+    InvalidObjectData, exception_handler,
+    InvalidObjectIdentifier,
+    InvalidResponseError,
+    InvalidURI,
+)
 
 logger = logging.getLogger(__name__)
+sys.excepthook = exception_handler
```

Comment:
Why is this required? 😅

Reply:
I didn't like that the program prints a traceback whenever any exception occurs. Exceptions that are raised for clearly defined reasons (e.g., the user enters a wrong URL etc.) should not print a traceback, just some informative error message. To make sure we are in full control of how errors are handled I have defined our own error handler